### PR TITLE
[TESTING, CYPRESS] 3624 tests for logs and class statistics button in class page

### LINKED ISSUE #3624: [TESTING] Create front-end tests to go stats and logs page
Front-end tests for checking the class statistics button and logs button in the `/for-teachers/class/[class id]` page.

### PR DESCRIPTION
**Pull request by hanna1231**
**Description**
With this test the logs and class statistics button in the `/for-teachers/class/[class id]` page. This test will also navigate to the `/logs/class/[class id]` and `/stats/class/[class id]` respectively and therefore this tests can be used for other front-tests.

**Fixes _issue or discussion number_**
Fixes #3624 

**How to test**
Run 
`export ADMIN_USER=admin_user; python app.py` in your virtual environment. This ensures that there is an active admin_user and Hedy is running.

Then you can go to the /tests folder and run: 

`npx cypress run --spec cypress/e2e/teacher_page/class_page`

### CODE REVIEW

File: tests/cypress/e2e/teacher_page/class_page/logs_button.cy.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  it('Passes', () => {
+    loginForTeacher();
+    cy.wait(500);
+    createClass();
+    cy.get(":nth-child(1) > :nth-child(3) > .no-underline").click(); // Press view class button
+
+    var currentUrl = '';
+    cy.url().then(url => {
+      currentUrl = url;
+      let logsButton = `[onclick="window.location.href = '/logs/class/` + currentUrl.substring(currentUrl.indexOf('class/')+6) + `'"]`; // Get logs button
```
Comment (TiBiBa) [inline tests/cypress/e2e/teacher_page/class_page/logs_button.cy.js:15]: Same here, what is we decide to change the url of the class? Is there a way to make this more flexible as well?

File: tests/cypress/e2e/teacher_page/class_page/logs_button.cy.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import {loginForAdmin, loginForTeacher} from '../../tools/login/login.js'
+import {createClass} from '../../tools/classes/class.js'
+
+
+describe('Is able to go back to teacher page', () => {
+  it('Passes', () => {
+    loginForTeacher();
+    cy.wait(500);
+    createClass();
+    cy.get(":nth-child(1) > :nth-child(3) > .no-underline").click(); // Press view class button
```
[the remark this answers is not in the document]
Comment (hanna1231) [reply tests/cypress/e2e/teacher_page/class_page/logs_button.cy.js:10]: I am not sure if this is something that I can make more flexible. If I for example would select buttons by using the text on it then it wouldn't work for other languages. I will have another look at it to see if I can find a solution but I saw that there was already a PR approved that also uses this method of selecting buttons.

**Comment by hanna1231** [comment]
I created id's to make selecting the buttons more flexible. I will also try to find a similar solution to making the url's more flexible

**Review by TiBiBa — CHANGES_REQUESTED** [review]
Running the tests seem to fail on the following issue :
```
 AssertionError: Timed out retrying after 4000ms: Expected to find element: `#teacher_classes > .green-btn`, but never found it.
```

I would propose to change this structure to adding an id to the "create class" button and calling it directly. Not sure if this would solve the issue, but would be a bit more flexible for future UI changes (adding buttons to the for-teachers page for example).

**Comment by hanna1231** [comment]
> 

I did not get this error myself which is strange but I followed your suggestion and hopefully it works now!

**Review by TiBiBa — APPROVED** [review]
All works, good job!

**Pull request merged**